### PR TITLE
chore: transform `url` arg into a flag

### PR DESCRIPTION
**Pull request by leovct**
# Description

**🚨 Breaking change 🚨**

It's just a suggestion, no problem if we don't merge it :)

This pull request enhances the `loadtest`, `loadtest uniswapv3`, `monitor` and `rpcfuzz` commands by transforming the `url` argument into a more flexible and user-friendly flag called `rpc-url`. With this update, users can now set the URL as a flag, offering greater control and simplicity in command usage, following the standards employed by other tools such as [cast](https://book.getfoundry.sh/cast/).

By default, the `rpc-url` flag is set to `http://localhost:8545`.

```sh
$ polycli loadtest --help | grep rpc-url
  -r, --rpc-url string                        The RPC endpoint url (default "http://localhost:8545")
```

It's now possible to run `polycli loadtest`. If you were specifying the url as an argument previously, you can switch from `polycli loadtest <url>` to `polycli loadtest --rpc-url <url>` or `polycli loadtest -r <url>`.

## Other

- Move most of the logic related to `cobra.Command` to a separate file `cmd.go` or `app.go` for each cmd.
- Move logic related to flags in `checkFlags` and logic responsible for running the cmd in a specific func.
- Add a method to validate urls in `util`, to be reused in other commands also.

For example, here's how the `monitor` cmd looks now.

```go
// cmd/monitor/cmd.go

// MonitorCmd represents the monitor command
var MonitorCmd = &cobra.Command{
	Use:   "monitor",
	Short: "Monitor blocks using a JSON-RPC endpoint.",
	Long:  usage,
	Args:  cobra.NoArgs,
	PreRunE: func(cmd *cobra.Command, args []string) error {
		return checkFlags()
	},
	RunE: func(cmd *cobra.Command, args []string) error {
		return monitor(cmd.Context())
	},
}
```

## Jira / Linear Tickets

- [DVT-1020](https://polygon.atlassian.net/browse/DVT-1020)

# Testing

x

[DVT-1020]: https://polygon.atlassian.net/browse/DVT-1020?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by leovct** [comment]
If we end up merging this PR, we should make sure we update the `polycli loadtest uniswapv3` cmd (see #137)

**Comment by leovct** [comment]
> If we end up merging this PR, we should make sure we update the `polycli loadtest uniswapv3` cmd (see #137)

Implemented in the PR ✅

File: cmd/loadtest/uniswapv3.go [hunk 1]
```diff
@@ -79,24 +62,6 @@ func checkFlags() error {
...
-		return nil, errors.New("the scheme has not been specified")
-	}
-	switch url.Scheme {
-	case "http", "https", "ws", "wss":
-		return url, nil
-	default:
-		return nil, fmt.Errorf("the scheme %s is not supported", url.Scheme)
-	}
-}
-
```
Comment (rebelArtists) [inline cmd/loadtest/uniswapv3.go:99]: nice simplification here

File: doc/polycli_loadtest.md [hunk 2]
```diff
@@ -14,7 +14,7 @@
 Run a generic load test against an Eth/EVM style JSON-RPC endpoint.
 
 ```bash
-polycli loadtest url [flags]
+polycli loadtest [flags]
```
Comment (rebelArtists) [inline doc/polycli_loadtest.md:17]: nit: potentially add comment here that rpc-url default is localhost

File: doc/polycli_loadtest_uniswapv3.md [hunk 3]
```diff
@@ -98,6 +98,7 @@ The command also inherits flags from parent commands.
       --private-key string                         The hex encoded private key that we'll use to send transactions (default "42b6e34dc21598a807dc19d7784c71b2a7a01f6480dc6f58258f78e539f1a1fa")
       --rate-limit float                           An overall limit to the number of requests per second. Give a number less than zero to remove this limit all together (default 4)
   -n, --requests int                               Number of requests to perform for the benchmarking session. The default is to just perform a single request which usually leads to non-representative benchmarking results. (default 1)
+  -r, --rpc-url string                             The RPC endpoint url (default "http://localhost:8545")
```
Comment (rebelArtists) [inline doc/polycli_loadtest_uniswapv3.md:101]: nvm, this is perfect

File: scripts/clients.mk [hunk 4]
```diff
@@ -37,10 +37,10 @@ geth-loadtest: build fund ## Run loadtest against an EVM/Geth chain.
 	sleep 5
 	$(BUILD_DIR)/$(BIN_NAME) loadtest \
 		--verbosity 700 \
+		--rpc-url http://127.0.0.1:$(PORT) \
```
Comment (rebelArtists) [inline scripts/clients.mk:40]: nit: potentially make host configurable

**Review by rebelArtists — APPROVED** [review]
looks great, couple nits but not a blocker

File: scripts/clients.mk (hunk 4 above)
Comment (leovct) [reply scripts/clients.mk:40]: yeah we could, this is just a handy command to trigger load test in ci :)

**Pull request merged**